### PR TITLE
Remove unneeded sponsor chain after 2nd device login

### PR DESCRIPTION
**Pull request by djlmills**

### CODE REVIEW

File: packages/interbit-ui-tools/src/middleware/interbitGlobal.js [hunk 1]
```diff
@@ -27,7 +27,25 @@ const createContext = async () => {
...
+    version: interbit.VERSION,
+    chainId: hypervisor.chainId
+  })
+
+  if (!hypervisorChainId) {
+    await localDataStore.setItem(
+      DATASTORE_KEYS.HYPERVISOR_CHAIN_ID,
+      hypervisor.chainId
+    )
+  }
```
Comment (djlmills) [inline packages/interbit-ui-tools/src/middleware/interbitGlobal.js]: This change means the browser hypervisor can load its chain state from local storage.

File: packages/interbit-ui-tools/src/middleware/reducer.js [hunk 2]
```diff
@@ -64,6 +64,10 @@ const reducer = (state = initialState, action = {}) => {
       return state.setIn([PUBLIC_KEY], publicKey)
     }
 
+    case actionTypes.INTERBIT_READY: {
+      return state.setIn([STATUS], INTERBIT_STATUS.READY)
+    }
+
```
Comment (djlmills) [inline packages/interbit-ui-tools/src/middleware/reducer.js]: This action didn't do anything before. Joy of unit tests.

File: packages/interbit-ui-tools/src/tests/middleware/reducer.test.js [hunk 3]
```diff
@@ -4,37 +4,246 @@ const { actionCreators } = require('../../middleware/actions')
...
-    const action1 = actionCreators.chainUpdated(chainName1, chainState1)
-    const action2 = actionCreators.chainUpdated(chainName2, chainState2)
+    assert.deepEqual(result.chains[chainAlias1], chainState1)
+    assert.deepEqual(result.chains[chainAlias2], chainState2)
+  })
 
-    const state = reducer(initialState, action1)
-    const result = reducer(state, action2)
+  it('updates existing chain state', () => {
+    const chainState1 = { ramen: 'meh' }
```
Comment (nicholeous) [inline packages/interbit-ui-tools/src/tests/middleware/reducer.test.js]: Ramen is amazing. Don't you slander noodles like that. :|

File: packages/interbit-ui-tools/src/tests/middleware/reducer.test.js [hunk 4]
```diff
@@ -4,37 +4,246 @@ const { actionCreators } = require('../../middleware/actions')
...
+    const action1 = actionCreators.chainUpdated(chainAlias, chainState1)
+    const intermediateState = reducer(initialState, action1)
+
+    assertExpectedState(intermediateState, {
+      expectedStateChange: { chains: { [chainAlias]: chainState1 } }
+    })
 
-    assert.deepEqual(result.chains[chainName1], chainState1)
-    assert.deepEqual(result.chains[chainName2], chainState2)
+    const chainState2 = { ramen: ['noodles', 'soup', 'love'] }
```
Comment (nicholeous) [inline packages/interbit-ui-tools/src/tests/middleware/reducer.test.js]: :joy:

File: packages/interbit-ui-tools/src/tests/middleware/reducer.test.js [hunk 5]
```diff
@@ -4,37 +4,246 @@ const { actionCreators } = require('../../middleware/actions')
...
+    const startState = initialState.merge({
+      chains: { [chainAlias]: { thingy: 'wotsit' } },
+      chainData: { [chainAlias]: { chainId, status: 'DELETING' } }
+    })
+    const action = actionCreators.chainDeleted({ chainAlias, chainId })
+    const result = reducer(startState, action)
+    assertExpectedState(result, {
+      startState,
+      expectedEndState: initialStateWithChains
+    })
```
Comment (nicholeous) [inline packages/interbit-ui-tools/src/tests/middleware/reducer.test.js]: :+1: For *ALL* the tests.  Very nice.

**Pull request merged**